### PR TITLE
:rocket: integrate with tsdownsample

### PR DESCRIPTION
**Pull request by jvdd**
- [x] avoid creating tsdownsample objects every time we downsample - should be done in the constructor
- [x] fix serialization -> addressed in https://github.com/predict-idlab/tsdownsample/pull/34
- [x] redefine the size-threshold, given that minmax is now way faster! 🐎 
- [x] verify docs
- [x] remove MinMaxOverlap?
  > I would keep it, but add comments about this aggregator.
- [x] remove lttbc & the build / setup.py commands
- [x] update the `README.md` with the `tsdownsample` integration! 🔥 
- [x] what about the `parallell` parameter?
  > would add the `**downsample_kwargs` argument to the datapoint selector impelemntations.
- [x] Verify that basic-example notebooks are still working.

---

Copy all Python implementation to tsdownsample - where we can properly test whether these yield the same results

### CODE REVIEW

File: plotly_resampler/aggregation/aggregators.py [hunk 1]
```diff
@@ -273,18 +248,13 @@ def _arg_downsample(
...
-
+        downsampler = LTTBDownsampler()
         if y.shape[0] > size_threshold and y.shape[0] / n_out > ratio_threshold:
-            # TODO: add argument for 30 when the paper is published
-            idxs = self.minmax._arg_downsample(x, y, n_out * 30)
-            y = y[idxs]
-            if x is not None:
-                x = x[idxs]
-        return self.lttb._arg_downsample(x, y, n_out)
+            downsampler = MinMaxLTTBDownsampler()
```
Comment (jonasvdd) [inline plotly_resampler/aggregation/aggregators.py:253]: Remains `MinMaxLTTB` after the y-shape condition was once met?

File: plotly_resampler/aggregation/aggregators.py [hunk 2]
```diff
@@ -198,38 +202,9 @@ def _arg_downsample(
...
-        #     (block_size - 1)
-        #     - np.fliplr(
-        #         s[: block_size * offset.shape[0]].values.reshape(-1, block_size)
-        #     ).argmax(axis=1)
-        # ) + offset
-
-        # Sort the argmin & argmax (where we append the first and last index item)
-        return np.unique(np.concatenate((argmin, argmax, [0, y.shape[0] - 1])))
+        if x is None:
+            return MinMaxDownsampler().downsample(y, n_out=n_out)
```
Comment (jonasvdd) [inline plotly_resampler/aggregation/aggregators.py:206]: This is not a MinMax Overlap Aggregator.

Comment (jvdd) [reply plotly_resampler/aggregation/aggregators.py:206]: Should be MinMax aggregator

File: pyproject.toml [hunk 3]
```diff
@@ -49,6 +49,7 @@ orjson = "^3.8.0"  # Faster json serialization
 Flask-Cors = { version = "^3.0.10", optional = true }
 # Lock kaleido dependency until https://github.com/plotly/Kaleido/issues/156 is resolved
 kaleido = {version = "0.2.1", optional = true}
+tsdownsample = "^0.1.1"
```
Comment (jonasvdd) [inline pyproject.toml:52]: 🚀

File: plotly_resampler/aggregation/aggregators.py (hunk 2 above)
Comment (jonasvdd) [reply plotly_resampler/aggregation/aggregators.py:206]: whoops. my bad.

File: plotly_resampler/aggregation/aggregators.py [hunk 4]
```diff
@@ -273,18 +248,13 @@ def _arg_downsample(
...
-            idxs = self.minmax._arg_downsample(x, y, n_out * 30)
-            y = y[idxs]
-            if x is not None:
-                x = x[idxs]
-        return self.lttb._arg_downsample(x, y, n_out)
+            downsampler = MinMaxLTTBDownsampler()
+
+        if x is None:
+            return downsampler.downsample(y, n_out=n_out)
+        return downsampler.downsample(x, y, n_out=n_out)
```
Comment (jonasvdd) [inline plotly_resampler/aggregation/aggregators.py:257]: What do we do with argument such as parallel?

Comment (jvdd) [reply plotly_resampler/aggregation/aggregators.py:251]: kwargs?

Comment (jonasvdd) [reply plotly_resampler/aggregation/aggregators.py:251]: maybe `**downsample_kwargs` then?

File: plotly_resampler/aggregation/aggregators.py [hunk 5]
```diff
@@ -90,7 +91,8 @@ def _arg_downsample(
         **_,
     ) -> np.ndarray:
         # Use the Core interface to perform the downsampling
-        return LTTB_core.downsample(x, y, n_out)
+        # return LTTB_core.downsample(x, y, n_out)
+        return LTTBDownsampler().downsample(x, y, n_out=n_out)
 
 
 class MinMaxOverlapAggregator(DataPointSelector):
```
Comment (jvdd) [inline plotly_resampler/aggregation/aggregators.py:98]: Do we remove this one? - I do not really see the point of keeping this one...

Comment (jonasvdd) [reply plotly_resampler/aggregation/aggregators.py:99]: ineed, I would remove the `LTTB_core`

File: docs/sphinx/conf.py [hunk 6]
```diff
@@ -65,8 +65,10 @@
 
 todo_include_todos = True
 
-autoclass_content = "both"
+# autoclass_content = "both"
 autodoc_typehints = "description"
+# Don't show class signature with the class' name.
+autodoc_class_signature = "separated"
```
Comment (jvdd) [inline docs/sphinx/conf.py:71]: Can you just add some minor comments on why this is changed

File: plotly_resampler/aggregation/plotly_aggregator_parser.py [hunk 7]
```diff
@@ -21,6 +21,8 @@ def parse_hf_data(
         #       pd.Categorical
         #   - pd.CategoricalIndex -> calling .values returns a pd.Categorical
         #   - pd.Categorical: has no .values attribute -> will not be parsed
+        if isinstance(hf_data, pd.RangeIndex):
+            return None
```
Comment (jvdd) [inline plotly_resampler/aggregation/plotly_aggregator_parser.py:25]: :fire:

File: plotly_resampler/aggregation/aggregators.py [hunk 8]
```diff
@@ -16,17 +16,23 @@
...
-    import warnings
 
-    warnings.warn("Could not import lttbc; will use a (slower) python alternative.")
-    from .algorithms.lttb_py import LTTB_core_py as LTTB_core
+def _to_tsdownsample_args(
+    x: np.ndarray | None, y: np.ndarray
+) -> Tuple[np.ndarray, ...]:
+    """Converts x & y to the arguments expected by tsdownsample."""
+    if x is None:
+        return (y,)
```
Comment (jvdd) [inline plotly_resampler/aggregation/aggregators.py:34]: :fire:

File: plotly_resampler/aggregation/aggregators.py [hunk 9]
```diff
@@ -62,6 +69,7 @@ class LTTB(DataPointSelector):
         >>> s = pd.Series(["a", "b", "c", "a"])
         >>> cat_type = pd.CategoricalDtype(categories=["b", "c", "a"], ordered=True)
         >>> s_cat = s.astype(cat_type)
+    * `LTTB` has no downsample kwargs, as it cannot be paralellized.
```
Comment (jvdd) [inline plotly_resampler/aggregation/aggregators.py:72]: Maybe plug here MinMaxLTTB?

**Review by jvdd — COMMENTED** [review]
Looks good! See comments

File: plotly_resampler/aggregation/aggregators.py (hunk 9 above)
Comment (jonasvdd) [reply plotly_resampler/aggregation/aggregators.py:72]: agree!

**Pull request merged**